src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -2383,6 +2405,10 @@ bool LibraryCallKit::inline_vector_convert() {
     op = gvn().transform(new VectorReinterpretNode(op, src_type, dst_type));
   }
 
+  if (is_vector_shuffle(vbox_klass_to)) {
+     op = wrap_indexes(op, num_elem_to, elem_bt_to);
+  }
+
```

Comment:
The wrap_indexes is needed only for two vector rearrange. It looks to me that doing a wrap_indexes here at convert would force it for single vector rearrange (or selectFrom) and thereby reduce the performance for that case as well. Please note that the single vector rearrange throws "IndexOutOfBoundsException" and doesn't need to do a wrap.

Reply:
Please ignore the above comment. I verified that each index is partially wrapped as part of toShuffle().  We should name the wrap_indexes() to partially_wrap_indexes() for clarity.